app/src/main/java/com/epfl/beatlink/ui/authentication/ProfileBuild.kt

```diff
@@ -0,0 +1,315 @@
+import com.epfl.beatlink.R
+import com.epfl.beatlink.model.authentication.FirebaseAuthViewModel
+import com.epfl.beatlink.ui.components.PrincipalButton
+import com.epfl.beatlink.ui.navigation.NavigationActions
+import com.epfl.beatlink.ui.navigation.Screen
+import com.epfl.beatlink.ui.theme.PrimaryGradientBrush
+import com.epfl.beatlink.ui.theme.SecondaryGray
+
+@SuppressLint("MutableCollectionMutableState")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ProfileBuildScreen(
+    navigationActions: NavigationActions,
+    firebaseAuthViewModel: FirebaseAuthViewModel =
+        viewModel(factory = FirebaseAuthViewModel.Factory)
+) {
+  var name by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+  var favoriteGenres by remember { mutableStateOf(mutableListOf<String>()) }
+  var isGenreSelectionVisible by remember { mutableStateOf(false) }
+
+  Scaffold(
+      modifier = Modifier.testTag("profileBuildScreen"),
+      topBar = { TopAppBar(title = { BeatLinkTopLogo() }) }) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.padding(paddingValues)
+                    .padding(16.dp)
+                    .verticalScroll(rememberScrollState()),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              // Profile Build Title Section
+              ProfileBuildTitle()
+
+              // Add Profile Picture
+              AddProfilePicture()
+
+              Spacer(modifier = Modifier.height(16.dp))
+
+              // Name input field
+              CustomInputField(
+                  value = name,
+                  onValueChange = { name = it },
+                  label = "Name",
+                  placeholder = "Enter name",
+                  keyboardType = KeyboardType.Email,
+                  modifier = Modifier.testTag("inputName"))
+
+              // Description input field
+              CustomInputField(
+                  value = description,
+                  onValueChange = { description = it },
+                  label = "Description",
+                  placeholder = "Enter description",
+                  supportingText = "max. 200 characters",
+                  modifier = Modifier.testTag("inputDescription"))
+
+              // Select favorite music genres
+              SelectFavoriteMusicGenres(
+                  favoriteGenres = favoriteGenres,
+                  onGenreSelectionVisibilityChanged = { isGenreSelectionVisible = it })
+
+              // Save button
+              PrincipalButton("Save", "saveButton") { navigationActions.navigateTo(Screen.HOME) }
+
+              // Show music genre selection dialog if visible
+              if (isGenreSelectionVisible) {
+                MusicGenreSelectionDialog(
+                    genres = listOf("Pop", "Rap", "Electro", "Rock", "Metal", "Jazz", "Classical"),
```

Comment:
As it is going to be used all across the app and it is sort of a "database" (i.e. the app suggests these categories) it will be useful to consider moving this to a centralized constant or repository  (for later)